internal/discovery/cache.go

```diff
@@ -47,39 +51,59 @@ func (c *Cache) Get(ctx context.Context, gvk schema.GroupVersionKind) (proto.Sch
 		if c.current == nil || time.Since(c.lastFill) > time.Hour*24 {
 			logger.V(0).Info("filling discovery cache")
 			if err := c.fillUnlocked(ctx); err != nil {
-				return nil, err
+				return nil, nil, err
 			}
 		}
 
-		model, ok := c.current[gvk]
+		model, ok := c.typeNameMap[gvk]
 		if !ok && c.fillWhenNotFound {
 			c.current = nil // invalidate cache - retrieve fresh schema on next attempt
-			discoveryCacheChanges.Inc()
```

Comment:
We don't want to track this anymore?

Reply:
I removed the metric since it doesn't seem that useful? We already log the same signal